core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1730,6 +1731,7 @@ public ServerBuilder requestIdGenerator(Supplier<? extends RequestId> requestIdS
     public ServerBuilder requestIdGenerator(
             Function<? super RoutingContext, ? extends RequestId> requestIdGenerator) {
         this.requestIdGenerator = requireNonNull(requestIdGenerator, "requestIdGenerator");
```

Comment:
Because we set the `requestIdGenerator` to the `virtualHostTemplate`, we don't need `requestIdGenerator` field in `ServerBuilder` anymore.
```suggestion
```
We also don't need `ServerConfig.requestIdGenerator()` anymore because we are going to add it to `VirtualHost.requestIdGenerator()`. However, it breaks the API, so we need to do:
- Deprecate `ServerConfig.requestIdGenerator()`
- Remove `requestIdGenerator` in `ServerBuilder` and get it from the default virtual host when creating `ServerConfig` using package-private getter.